File: 2864-maximum-odd-binary-number/2864-maximum-odd-binary-number.cpp
class Solution {
public:
    string maximumOddBinaryNumber(string s) {
        int n = s.size();
        int countOnes = 0;
        for (char& c: s) {
            countOnes += c == '1';
        }
        for (int i = 0; i < countOnes - 1; i++) {
            s[i] = '1';
        }
        for (int i = countOnes - 1; i < n - 1; i++) {
            s[i] = '0';
        }
        s[n - 1] = '1';
        return s;
    }
};

File: 2864-maximum-odd-binary-number/2864-maximum-odd-binary-number.java
class Solution {
    public String maximumOddBinaryNumber(String s) {
        int oneCount = 0;
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '1') {
                oneCount += 1;
            }
        }
        StringBuilder builder = new StringBuilder();
        for (int i = 0; i < oneCount - 1; i++) {
            builder.append('1');
        }
        for (int i = oneCount; i < s.length(); i++) {
            builder.append('0');
        }
        builder.append('1');
        return builder.toString();
    }
}